homeassistant/components/ezviz/select.py

```diff
@@ -21,19 +27,105 @@
+            SoundMode, ezvizSelect.data[ezvizSelect.entity_description.key]
+        ).value
+        if sound_mode_value in [0, 1, 2]:
+            return ezvizSelect.options[sound_mode_value]
+
+        return None
+
+    @staticmethod
+    def select_option(ezvizSelect: EzvizSelect, serial: str, option: str) -> None:
+        """Change the selected option."""
+        sound_mode_value = ezvizSelect.options.index(option)
+
+        try:
+            ezvizSelect.coordinator.ezviz_client.alarm_sound(
+                serial, sound_mode_value, 1
+            )
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Cannot set Warning sound level for {ezvizSelect.entity_id}"
+            ) from err
+
+
+class BatteryWorkModeAction(EzvizSelectEntityActionBase):
+    """Class dedicated to Battery Work Mode."""
+
+    @staticmethod
+    def current_option(ezvizSelect: EzvizSelect) -> str | None:
+        """Return the selected entity option to represent the entity state."""
+        battery_work_mode = getattr(
+            BatteryCameraWorkMode,
+            ezvizSelect.data[ezvizSelect.entity_description.key],
+            BatteryCameraWorkMode.UNKNOWN,
+        )
+        if battery_work_mode == BatteryCameraWorkMode.UNKNOWN:
+            return None
+
+        return battery_work_mode.name.lower()
+
+    @staticmethod
+    def select_option(ezvizSelect: EzvizSelect, serial: str, option: str) -> None:
+        """Change the selected option."""
+        battery_work_mode = getattr(BatteryCameraWorkMode, option.upper())
+        try:
+            ezvizSelect.coordinator.ezviz_client.set_battery_camera_work_mode(
+                serial, battery_work_mode.value
+            )
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Cannot set battery work mode for {ezvizSelect.entity_id}"
+            ) from err
+
+
 @dataclass(frozen=True, kw_only=True)
 class EzvizSelectEntityDescription(SelectEntityDescription):
     """Describe a EZVIZ Select entity."""
 
     supported_switch: int
+    action_handler: type[EzvizSelectEntityActionBase]
 
 
-SELECT_TYPE = EzvizSelectEntityDescription(
+ALARM_SOUND_MODE_SELECT_TYPE = EzvizSelectEntityDescription(
     key="alarm_sound_mod",
     translation_key="alarm_sound_mode",
     entity_category=EntityCategory.CONFIG,
     options=["soft", "intensive", "silent"],
     supported_switch=DeviceSwitchType.ALARM_TONE.value,
+    action_handler=AlarmSoundModeAction,
+)
+
+BATTERY_WORK_MODE_SELECT_TYPE = EzvizSelectEntityDescription(
+    key="battery_camera_work_mode",
+    translation_key="battery_camera_work_mode",
+    icon="mdi:battery-sync",
+    entity_category=EntityCategory.CONFIG,
+    options=[
+        "plugged_in",
+        "high_performance",
+        "power_save",
+        "super_power_save",
+        "custom",
+    ],
+    supported_switch=-1,
+    action_handler=BatteryWorkModeAction,
```

Comment:
```suggestion
    current_option_fn=current_battery_work_mode,
    select_option_fn=lambda client, serial_no, option: client.set_battery_camera_work_mode(
                serial, BatteryCameraWorkMode(option)
            )
```
Where `current_battery_work_mode` is the func I was talking about